content/courses/advanced-ada/chapters/numerics.rst

```diff
@@ -1639,12 +1639,747 @@ precision, 0.2774 is rounded to 0.277, and 0.2777 is rounded to 0.278.
 Big Numbers
 -----------
 
+As we've seen before, we can define numeric types in Ada with a high degree of
+precision. However, in certain applications, even that precision isn't enough,
+so we have to rely on
+:wikipedia:`arbitrary-precision_arithmetic <arbitrary-precision_arithmetic>`.
+Ada's big numbers allow us to operate with this kind of arithmetic.
+
+Ada supports two categories of big numbers: big integers and big reals |mdash|
+both are specified in child packages of the :ada:`Ada.Numerics.Big_Numbers`
+package:
+
++--------------+----------------------------------------------+
+| Category     | Package                                      |
++==============+==============================================+
+| Big Integers | :ada:`Ada.Numerics.Big_Numbers.Big_Integers` |
++--------------+----------------------------------------------+
+| Big Reals    | :ada:`Ada.Numerics.Big_Numbers.Big_Real`     |
++--------------+----------------------------------------------+
+
 .. admonition:: Relevant topics
 
     - :arm22:`Big Numbers <A-5-5>`
     - :arm22:`Big Integers <A-5-6>`
     - :arm22:`Big Reals <A-5-7>`
 
-.. todo::
+Overview
+~~~~~~~~
+
+Let's start with a simple declaration of big numbers:
+
+.. code:: ada run_button project=Courses.Advanced_Ada.Numerics.Simple_Big_Numbers
+
+    pragma Ada_2022;
+
+    with Ada.Text_IO; use Ada.Text_IO;
+
+    with Ada.Numerics.Big_Numbers.Big_Integers;
+    use  Ada.Numerics.Big_Numbers.Big_Integers;
+
+    with Ada.Numerics.Big_Numbers.Big_Reals;
+    use  Ada.Numerics.Big_Numbers.Big_Reals;
+
+    procedure Show_Simple_Big_Numbers is
+       BI : Big_Integer;
+       BR : Big_Real;
+    begin
+       BI := 12345678901234567890;
+       BR := 2.0 ** 1234;
+
+       Put_Line ("BI: " & BI'Image);
+       Put_Line ("BR: " & BR'Image);
+
+       BI := @ + 1;
+       BR := @ + 1.0;
+
+       Put_Line ("BI: " & BI'Image);
+       Put_Line ("BR: " & BR'Image);
+    end Show_Simple_Big_Numbers;
+
+In this example, we're declaring the big integer :ada:`BI` and the big real
+:ada:`BR`, and we're incrementing them with one.
+
+Naturally, we're not limited to using the :ada:`+` operator (such as in this
+example). We can use any operators on big numbers in the same as we do with
+integers and floating-point variables. In fact, the common unary operators
```

Comment:
It might be simpler to just say "We can use the same operators on big numbers that we can use with normal numeric types." which would be better as well because it isn't only variables that are involved. It's a matter of type compatibility.

Reply:
Yes, your version of the phrase is simpler and more precise.